# 04 – Dev-Abschlusslauf: Finale Konfiguration (min_on=1.0, min_off=1.2, beam=12, len=20)

## Kontext

Dieses Notebook ist der **finale Dev-Durchlauf** mit der in `02k_` identifizierten
optimalen Parameterkombination:

| Parameter | Wert | Quelle |
|-----------|------|--------|
| `beam_size` | 12 | `02_` + `02j_` Grid-Search (Decoding) |
| `max_length` | 20 s | `02_` + `02j_` Grid-Search (Decoding) |
| `min_duration_on` | 1.0 s | `02k_` Grid-Search (E72) |
| `min_duration_off` | 1.2 s | `02k_` Grid-Search (E72) |
| Modell | BL4 (`avsr_cocktail_finetuned`) | Baseline |

## Ablauf

1. Segmentierungs-Monkey-Patch (aus `02k_`) aktiv setzen
2. Inference auf allen Dev-Sessions
3. Evaluation + Vergleich mit BL4-Baseline

## Ergebnis (Vorschau)

| Konfiguration | WER | Conv F1 | Joint Error | Δ WER vs. BL4 |
|--------------|-----|---------|-------------|---------------|
| BL4-Baseline (beam=3, len=15) | 0.4987 | 0.8153 | 0.3547 | – |
| **Dev-Final (E72-Konfig, 25 Sessions)** | **0.4943** | **0.8553** | **0.3318** | **-0.0044 (-0.88%)** |

Bestes Ergebnis der gesamten Experimentreihe auf dem Dev-Set.

## 1 – GPU-Check

In [1]:
!nvidia-smi

Thu Feb  5 04:52:48 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.54.15              Driver Version: 550.54.15      CUDA Version: 12.5     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A100-SXM4-80GB          On  |   00000000:01:00.0 Off |                    0 |
| N/A   33C    P0             80W /  500W |    9557MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+-----

## 2 – CUDA-Verifikation

In [2]:
import os

# Physische GPU-Auswahl: hier GPU 2 (siehe nvidia-smi)
os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
os.environ["CUDA_VISIBLE_DEVICES"] = "1"


In [3]:
import torch, gc, glob
from pathlib import Path
import pandas as pd

In [4]:
print("CUDA devices:", torch.cuda.device_count())
print("Device 0 name (sollte A100 sein):", torch.cuda.get_device_name(0))

CUDA devices: 1
Device 0 name (sollte A100 sein): NVIDIA A100-SXM4-80GB


## 3 – Setup: Arbeitsverzeichnis & Imports

In [5]:
project_baseline_path = "/home/josch080/Projektgruppe/mcorec_baseline"
os.chdir(project_baseline_path)

import sys
if project_baseline_path not in sys.path:
    sys.path.append(project_baseline_path)

from script.pg_utils_experiments import run_inference_for_experiment, append_eval_results_for_experiments

/home/josch080/Projektgruppe/mcorec_baseline/script/inference.py:22: FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.
  if not hasattr(np, "object"):


## 4 – Segmentierungs-Patch (aus `02k_`)

Identisch zu `02k_`: Context-Manager überschreibt `InferenceEngine.chunk_video`
temporär, damit `min_duration_on` und `min_duration_off` an `segment_by_asd()`
weitergegeben werden. Nach dem `with`-Block wird die Original-Methode wiederhergestellt.

In [6]:
from contextlib import contextmanager
import script.inference as inf  # das ist das Modul, das pg_utils_experiments nutzt

@contextmanager
def patch_avsr_segmentation(min_on=None, min_off=None):
    # Patcht InferenceEngine.chunk_video für den ASD-Zweig
    # Wird als Context-Manager verwendet; Original wird in finally wiederhergestellt
    orig = inf.InferenceEngine.chunk_video

    if min_on is None and min_off is None:
        yield
        return

    def patched(self, video_path, asd_path=None, max_length=15):
        if asd_path is not None:
            with open(asd_path, "r") as f:
                asd = inf.json.load(f)

            frames = sorted([int(f) for f in asd.keys()])
            if not frames:
                return []
            min_frame = min(frames)

            params = {"max_chunk_size": max_length}
            if min_on is not None:
                params["min_duration_on"] = float(min_on)
            if min_off is not None:
                params["min_duration_off"] = float(min_off)

            segments_by_frames = inf.segment_by_asd(asd, params)
            return [((seg[0] - min_frame) / 25, (seg[-1] - min_frame) / 25) for seg in segments_by_frames]

        return orig(self, video_path, asd_path, max_length)

    inf.InferenceEngine.chunk_video = patched
    try:
        yield
    finally:
        inf.InferenceEngine.chunk_video = orig

def _tag(x: float) -> str:
    # Float → Dateinamen-sicherer String: 1.2 → '1p2'
    return str(x).replace(".", "p")

## 5 – Split-Konfiguration

`SPLIT = "DEV"` → `data-bin/dev`. Dasselbe Notebook kann für EVAL genutzt werden
(nur `SPLIT` ändern) – `04_eval_final_results` tut genau das.

In [7]:
SPLIT = "DEV"
DATA_ROOT = "data-bin/dev" if SPLIT == "DEV" else "data-bin/eval"  # ggf. anpassen

## 6 – Experiment-Definition

Automatisch generierter `exp_name` kodiert alle relevanten Parameter:
`DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20`.
Nur BL4 (`avsr_cocktail_finetuned`) im `BASE_MODELS`-Dict, da ausschließlich dieses Modell genutzt wird.

In [8]:
beam_size  = 12
max_length = 20
best_min_on  = 1.0 # Beste Konfiguration aus 02j_ (E72)
best_min_off = 1.2 # Beste Konfiguration aus 02j_ (E72)

# exp_name kodiert alle Parameter im Verzeichnisnamen
exp_name = f"{SPLIT}_final_bugfix_mdOn{_tag(best_min_on)}_mdOff{_tag(best_min_off)}_bs{beam_size}_len{max_length}"
BASE_MODELS = {
    "avsr_cocktail_finetuned": {
        "model_type": "avsr_cocktail",
        "chkpt": "model-bin/avsr_cocktail_mcorec_finetune",
    },
}
EXPERIMENTS = {
    exp_name: {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": beam_size,
        "max_length": max_length,
        "comment": f"{SPLIT} FINAL: AVSR override min_on={best_min_on}, min_off={best_min_off}",
    }
}

## 7 – Session-Erkennung

`metadata.json`-Filter stellt sicher, dass nur vollständig vorbereitete
Sessions verarbeitet werden (keine leeren oder unvollständigen Verzeichnisse).

In [12]:
# Sessions für den Split finden
session_dirs = sorted([p for p in glob.glob(f"{DATA_ROOT}/session_*")
                       if Path(p, "metadata.json").exists()])
session_ids = [Path(p).name for p in session_dirs]
print("Split:", SPLIT, "| Sessions:", len(session_ids))
print(session_ids)

Split: DEV | Sessions: 25
['session_132', 'session_133', 'session_134', 'session_135', 'session_136', 'session_137', 'session_138', 'session_139', 'session_140', 'session_141', 'session_40', 'session_41', 'session_42', 'session_43', 'session_44', 'session_48', 'session_49', 'session_50', 'session_51', 'session_52', 'session_53', 'session_54', 'session_55', 'session_56', 'session_57']


## 8 – Inference

Jede Session wird innerhalb des Monkey-Patch-Context-Managers verarbeitet.
CUDA-Cache und GC werden nach jeder Session geleert.

In [13]:
# ======= Inference pro Session =======
for session_dir in session_dirs:
    print("\nRUN:", SPLIT, Path(session_dir).name)
    with patch_avsr_segmentation(min_on=best_min_on, min_off=best_min_off):
        run_inference_for_experiment(
            exp_name=exp_name,
            base_models=BASE_MODELS,
            experiments=EXPERIMENTS,
            session_dir=session_dir,
        )
    torch.cuda.empty_cache() # GPU-Speicher nach jeder Session freigeben
    gc.collect()


RUN: DEV session_132

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_132
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...


/home/josch080/Projektgruppe/mcorec_train/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_132


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:46<08:53, 106.68s/it]


Processing speakers:  33%|███▎      | 2/6 [02:58<05:45, 86.44s/it] 


Processing speakers:  50%|█████     | 3/6 [04:39<04:39, 93.06s/it]


Processing speakers:  67%|██████▋   | 4/6 [05:24<02:27, 73.91s/it]


Processing speakers:  83%|████████▎ | 5/6 [06:21<01:07, 67.69s/it]


Processing speaker spk_5 track 1 of 5: 100%|██████████| 10/10 [00:27<00:00,  2.72s/it]

Processing speaker spk_5 track 2 of 5: 100%|██████████| 4/4 [00:04<00:00,  1.12s/it]

Processing speaker spk_5 track 3 of 5: 100%|██████████| 10/10 [00:21<00:00,  2.16s/it]

Processing speaker spk_5 track 4 of 5: 100%|██████████| 4/4 [00:05<00:00,  1.29s/it]

Processing speakers: 100%|██████████| 6/6 [07:20<00:00, 73.45s/it]



RUN: DEV session_133

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_133
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_133


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 20/20 [00:45<00:00,  2.27s/it]

Processing speakers:  17%|█▋        | 1/6 [00:56<04:42, 56.43s/it]


Processing speaker spk_1 track 1 of 2: 100%|██████████| 14/14 [00:30<00:00,  2.19s/it]

Processing speakers:  33%|███▎      | 2/6 [01:59<04:01, 60.28s/it]


Processing speakers:  50%|█████     | 3/6 [03:36<03:50, 76.98s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:13<02:02, 61.12s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:15<01:01, 61.53s/it]


Processing speakers: 100%|██████████| 6/6 [06:48<00:00, 68.15s/it]



RUN: DEV session_134

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_134
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_134


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:22<06:50, 82.09s/it]


Processing speakers:  33%|███▎      | 2/6 [02:29<04:53, 73.48s/it]


Processing speakers:  50%|█████     | 3/6 [03:36<03:31, 70.56s/it]


Processing speakers:  67%|██████▋   | 4/6 [05:01<02:32, 76.26s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:22<00:00,  1.42s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:57<01:08, 68.97s/it]


Processing speakers: 100%|██████████| 6/6 [06:53<00:00, 68.88s/it]



RUN: DEV session_135

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_135
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_135


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 15/15 [00:23<00:00,  1.55s/it]

Processing speakers:  20%|██        | 1/5 [01:01<04:05, 61.43s/it]


Processing speakers:  40%|████      | 2/5 [02:03<03:05, 61.85s/it]


Processing speakers:  60%|██████    | 3/5 [03:25<02:22, 71.10s/it]


Processing speakers:  80%|████████  | 4/5 [04:27<01:07, 67.40s/it]


Processing speakers: 100%|██████████| 5/5 [05:45<00:00, 69.20s/it]



RUN: DEV session_136

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_136
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_136


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:38<08:12, 98.45s/it]


Processing speakers:  33%|███▎      | 2/6 [02:32<04:50, 72.53s/it]


Processing speakers:  50%|█████     | 3/6 [03:24<03:08, 62.87s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:26<02:05, 62.71s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 19/19 [01:13<00:00,  3.86s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:52<01:11, 71.05s/it]


Processing speakers: 100%|██████████| 6/6 [06:31<00:00, 65.33s/it]



RUN: DEV session_137

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_137
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_137


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:36<03:04, 36.96s/it]


Processing speakers:  33%|███▎      | 2/6 [02:28<05:22, 80.69s/it]


Processing speakers:  50%|█████     | 3/6 [03:06<03:04, 61.34s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 20/20 [00:21<00:00,  1.07s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:56<01:53, 56.84s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:51<00:56, 56.01s/it]


Processing speakers: 100%|██████████| 6/6 [05:44<00:00, 57.37s/it]



RUN: DEV session_138

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_138
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_138


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:28<02:24, 28.87s/it]


Processing speakers:  33%|███▎      | 2/6 [02:08<04:43, 70.78s/it]


Processing speakers:  50%|█████     | 3/6 [02:56<03:00, 60.28s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 21/21 [00:48<00:00,  2.31s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:00<02:03, 61.76s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:37<00:52, 52.61s/it]


Processing speaker spk_5 track 1 of 4: 100%|██████████| 7/7 [00:07<00:00,  1.12s/it]

Processing speaker spk_5 track 2 of 4: 100%|██████████| 15/15 [00:30<00:00,  2.05s/it]

Processing speaker spk_5 track 3 of 4: 100%|██████████| 6/6 [00:06<00:00,  1.00s/it]

Processing speakers: 100%|██████████| 6/6 [05:24<00:00, 54.12s/it]



RUN: DEV session_139

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_139
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_139


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:23<06:57, 83.57s/it]


Processing speakers:  33%|███▎      | 2/6 [02:33<05:02, 75.68s/it]


Processing speakers:  50%|█████     | 3/6 [03:24<03:12, 64.08s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:24<00:00,  1.36s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:57<01:43, 51.91s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:45<00:50, 50.75s/it]


Processing speakers: 100%|██████████| 6/6 [05:46<00:00, 57.73s/it]



RUN: DEV session_140

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_140
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_140


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:51<04:17, 51.43s/it]


Processing speakers:  33%|███▎      | 2/6 [01:33<03:04, 46.05s/it]


Processing speakers:  50%|█████     | 3/6 [02:53<03:04, 61.63s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 5/5 [00:07<00:00,  1.58s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 10/10 [00:11<00:00,  1.11s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:35<01:47, 53.67s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:19<00:50, 50.33s/it]


Processing speaker spk_5 track 1 of 3: 100%|██████████| 6/6 [00:10<00:00,  1.67s/it]

Processing speaker spk_5 track 2 of 3: 100%|██████████| 5/5 [00:04<00:00,  1.08it/s]

Processing speakers: 100%|██████████| 6/6 [04:55<00:00, 49.32s/it]



RUN: DEV session_141

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_141
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_141


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speakers:  20%|██        | 1/5 [00:47<03:08, 47.09s/it]


Processing speakers:  40%|████      | 2/5 [01:55<02:58, 59.51s/it]


Processing speakers:  60%|██████    | 3/5 [02:50<01:54, 57.44s/it]


Processing speakers:  80%|████████  | 4/5 [03:24<00:48, 48.14s/it]


Processing speakers: 100%|██████████| 5/5 [04:11<00:00, 50.21s/it]



RUN: DEV session_40

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_40
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:44<03:42, 44.46s/it]


Processing speakers:  33%|███▎      | 2/6 [02:16<04:49, 72.50s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 13/13 [00:45<00:00,  3.50s/it]

Processing speakers:  50%|█████     | 3/6 [03:48<04:04, 81.57s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:45<00:00,  2.51s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:59<02:34, 77.16s/it]


Processing speakers:  83%|████████▎ | 5/6 [06:06<01:13, 73.50s/it]


Processing speakers: 100%|██████████| 6/6 [07:24<00:00, 74.02s/it]



RUN: DEV session_41

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_41
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_41


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 19/19 [00:24<00:00,  1.27s/it]

Processing speakers:  17%|█▋        | 1/6 [00:25<02:08, 25.76s/it]


Processing speaker spk_1 track 1 of 2: 0it [00:00, ?it/s]

Processing speakers:  33%|███▎      | 2/6 [02:18<05:07, 76.98s/it]


Processing speaker spk_2 track 1 of 2: 0it [00:00, ?it/s]

Processing speakers:  50%|█████     | 3/6 [03:50<04:11, 83.74s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 3/3 [00:10<00:00,  3.44s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:43<02:23, 71.55s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:45<01:08, 68.31s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 29/29 [00:58<00:00,  2.03s/it]

Processing speakers: 100%|██████████| 6/6 [06:45<00:00, 67.54s/it]



RUN: DEV session_42

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_42
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_42


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 12/12 [00:15<00:00,  1.30s/it]

Processing speakers:  17%|█▋        | 1/6 [00:50<04:14, 50.86s/it]


Processing speakers:  33%|███▎      | 2/6 [02:29<05:15, 78.98s/it]


Processing speakers:  50%|█████     | 3/6 [04:12<04:29, 89.74s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 4/4 [00:05<00:00,  1.38s/it]

Processing speakers:  67%|██████▋   | 4/6 [05:08<02:32, 76.40s/it]


Processing speakers:  83%|████████▎ | 5/6 [06:13<01:12, 72.46s/it]


Processing speakers: 100%|██████████| 6/6 [07:03<00:00, 70.64s/it]



RUN: DEV session_43

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_43
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [01:08<00:00,  2.55s/it]

Processing speakers:  17%|█▋        | 1/6 [01:28<07:23, 88.67s/it]


Processing speakers:  33%|███▎      | 2/6 [02:13<04:12, 63.05s/it]


Processing speakers:  50%|█████     | 3/6 [03:20<03:13, 64.57s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:42<02:22, 71.45s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 15/15 [00:26<00:00,  1.73s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:37<01:05, 65.72s/it]


Processing speakers: 100%|██████████| 6/6 [06:26<00:00, 64.43s/it]



RUN: DEV session_44

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_44
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_44


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [01:24<07:03, 84.69s/it]


Processing speakers:  33%|███▎      | 2/6 [02:05<03:55, 58.83s/it]


Processing speakers:  50%|█████     | 3/6 [03:05<02:58, 59.41s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 15/15 [00:23<00:00,  1.57s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:44<01:43, 51.51s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:37<00:51, 51.94s/it]


Processing speaker spk_5 track 1 of 3: 100%|██████████| 23/23 [00:22<00:00,  1.03it/s]

Processing speaker spk_5 track 2 of 3: 100%|██████████| 2/2 [00:00<00:00,  2.01it/s]

Processing speakers: 100%|██████████| 6/6 [05:07<00:00, 51.17s/it]



RUN: DEV session_48

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_48
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_48


Processing speakers:   0%|          | 0/3 [00:00<?, ?it/s]


Processing speaker spk_1 track 1 of 3: 100%|██████████| 2/2 [00:01<00:00,  1.54it/s]

Processing speaker spk_1 track 2 of 3: 100%|██████████| 6/6 [00:04<00:00,  1.40it/s]

Processing speakers:  33%|███▎      | 1/3 [00:54<01:49, 54.72s/it]


Processing speakers:  67%|██████▋   | 2/3 [01:28<00:42, 42.53s/it]


Processing speaker spk_3 track 1 of 10: 100%|██████████| 4/4 [00:02<00:00,  1.52it/s]

Processing speaker spk_3 track 2 of 10: 0it [00:00, ?it/s]

Processing speaker spk_3 track 3 of 10: 100%|██████████| 2/2 [00:01<00:00,  1.32it/s]

Processing speaker spk_3 track 4 of 10: 100%|██████████| 8/8 [00:17<00:00,  2.24s/it]

Processing speaker spk_3 track 5 of 10: 100%|██████████| 8/8 [00:16<00:00,  2.03s/it]

Processing speaker spk_3 track 6 of 10: 100%|██████████| 2/2 [00:02<00:00,  1.05s/it]

Processing speaker spk_3 track 7 of 10: 100%|██████████| 2/2 [00:06<00:00,  3.28s/it]

Processing speaker spk_3 track 8 of 10: 100%|██████████| 1/1 [00:10<00:00, 10.12s/it]

Processing speaker spk_3 track 9 of 10: 0it [00:00, ?it/s]

Processing speakers: 100%|██████████| 3/3 [02:32<00:00, 51.00s/it]



RUN: DEV session_49

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_49
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:15<01:18, 15.72s/it]


Processing speakers:  33%|███▎      | 2/6 [00:49<01:46, 26.60s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.75it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.87it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.11it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 3/3 [00:16<00:00,  5.50s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:08<00:00,  4.39s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:17<00:00,  2.20s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.06it/s]

Processing speakers:  50%|█████     | 3/6 [01:53<02:10, 43.63s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:39<01:28, 44.41s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:51<00:54, 54.46s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 20/20 [00:45<00:00,  2.26s/it]

Processing speakers: 100%|██████████| 6/6 [04:55<00:00, 49.18s/it]



RUN: DEV session_50

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_50
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:37<03:06, 37.22s/it]


Processing speakers:  33%|███▎      | 2/6 [01:23<02:50, 42.56s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 17/17 [00:21<00:00,  1.28s/it]

Processing speakers:  50%|█████     | 3/6 [01:53<01:50, 36.90s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:22<00:00,  1.40s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 11/11 [00:22<00:00,  2.06s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:58<01:35, 47.85s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:25<00:40, 40.54s/it]


Processing speakers: 100%|██████████| 6/6 [04:33<00:00, 45.54s/it]



RUN: DEV session_51

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_51
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_51


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:51<04:16, 51.31s/it]


Processing speaker spk_1 track 1 of 2: 100%|██████████| 14/14 [00:29<00:00,  2.14s/it]

Processing speakers:  33%|███▎      | 2/6 [01:57<04:01, 60.30s/it]


Processing speakers:  50%|█████     | 3/6 [02:39<02:35, 51.90s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:21<02:23, 71.66s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:09<01:02, 62.87s/it]


Processing speakers: 100%|██████████| 6/6 [05:42<00:00, 57.10s/it]



RUN: DEV session_52

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_52
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_52


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:39<03:19, 39.98s/it]


Processing speakers:  33%|███▎      | 2/6 [00:59<01:51, 27.87s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 2/2 [00:01<00:00,  1.35it/s]

Processing speakers:  50%|█████     | 3/6 [02:35<02:57, 59.04s/it]


Processing speakers:  67%|██████▋   | 4/6 [03:24<01:49, 54.96s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:44<00:42, 42.52s/it]


Processing speakers: 100%|██████████| 6/6 [05:41<00:00, 56.97s/it]



RUN: DEV session_53

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_53
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_53


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speakers:  20%|██        | 1/5 [01:18<05:14, 78.66s/it]


Processing speakers:  40%|████      | 2/5 [02:25<03:35, 71.80s/it]


Processing speakers:  60%|██████    | 3/5 [03:23<02:10, 65.45s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 20/20 [01:13<00:00,  3.67s/it]

Processing speakers:  80%|████████  | 4/5 [04:59<01:17, 77.50s/it]


Processing speakers: 100%|██████████| 5/5 [05:40<00:00, 68.14s/it]



RUN: DEV session_54

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_54
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 26/26 [00:46<00:00,  1.79s/it]

Processing speakers:  20%|██        | 1/5 [01:00<04:03, 60.93s/it]


Processing speakers:  40%|████      | 2/5 [02:21<03:37, 72.37s/it]


Processing speakers:  60%|██████    | 3/5 [03:17<02:09, 64.77s/it]


Processing speakers:  80%|████████  | 4/5 [04:41<01:12, 72.49s/it]


Processing speakers: 100%|██████████| 5/5 [05:26<00:00, 65.28s/it]



RUN: DEV session_55

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_55
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_55


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 22/22 [00:43<00:00,  1.98s/it]

Processing speakers:  20%|██        | 1/5 [01:13<04:52, 73.12s/it]


Processing speakers:  40%|████      | 2/5 [02:29<03:44, 74.85s/it]


Processing speakers:  60%|██████    | 3/5 [03:27<02:14, 67.15s/it]


Processing speakers:  80%|████████  | 4/5 [03:47<00:48, 48.76s/it]


Processing speakers: 100%|██████████| 5/5 [05:50<00:00, 70.00s/it]



RUN: DEV session_56

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_56
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_56


Processing speakers:   0%|          | 0/4 [00:00<?, ?it/s]


Processing speakers:  25%|██▌       | 1/4 [00:30<01:31, 30.53s/it]


Processing speakers:  50%|█████     | 2/4 [02:09<02:21, 70.85s/it]


Processing speakers:  75%|███████▌  | 3/4 [03:40<01:20, 80.02s/it]


Processing speakers: 100%|██████████| 4/4 [03:55<00:00, 58.78s/it]



RUN: DEV session_57

Starte Inference für Experiment: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
  session_dir     = data-bin/dev/session_57
  comment         = DEV FINAL: AVSR override min_on=1.0, min_off=1.2
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_57


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 21/21 [01:12<00:00,  3.47s/it]

Processing speakers:  20%|██        | 1/5 [01:34<06:17, 94.29s/it]


Processing speakers:  40%|████      | 2/5 [02:17<03:13, 64.48s/it]


Processing speakers:  60%|██████    | 3/5 [02:53<01:42, 51.24s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 21/21 [00:27<00:00,  1.30s/it]

Processing speakers:  80%|████████  | 4/5 [03:50<00:53, 53.66s/it]


Processing speakers: 100%|██████████| 5/5 [05:10<00:00, 62.19s/it]


## 9 – Evaluation & Aggregation

In [15]:
# ======= Evaluation + CSV =======
df_by_session = append_eval_results_for_experiments(
    experiments=EXPERIMENTS,
    session_ids=session_ids,
    target_csv=f"results_{exp_name}_by_session.csv",
    session_dir_template=f"{DATA_ROOT}" + "/{sid}",
    label_dir="labels",
)

summary = (df_by_session
           .groupby("model", as_index=False)[["avg_conv_f1","avg_speaker_wer","avg_joint_error"]]
           .mean())

summary.to_csv(f"results_{exp_name}_summary.csv", index=False)
display(summary)
print("Saved:", f"results_{exp_name}_by_session.csv", f"results_{exp_name}_summary.csv")


########## Evaluate für session_132 ##########
Starte Evaluate: /home/josch080/Projektgruppe/mcorec_train/bin/python script/evaluate.py --session_dir data-bin/dev/session_132 --output_dir_name output_ --label_dir_name labels
Evaluating 1 sessions

=== Evaluating session session_132 ===

--- Evaluating output dir: output_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20 ---
Conversation clustering F1 score: 1.0
Speaker to WER: {'spk_0': 0.3473, 'spk_1': 0.5906, 'spk_2': 0.4851, 'spk_3': 0.4842, 'spk_4': 0.6405, 'spk_5': 0.6696}
Speaker clustering F1 score: {'spk_0': 1.0, 'spk_1': 1.0, 'spk_2': 1.0, 'spk_3': 1.0, 'spk_4': 1.0, 'spk_5': 1.0}
Joint ASR-Clustering Error Rate: {'spk_0': 0.17365, 'spk_1': 0.2953, 'spk_2': 0.24255, 'spk_3': 0.2421, 'spk_4': 0.32025, 'spk_5': 0.3348}

--- Evaluating output dir: output_auto_avsr ---
Conversation clustering F1 score: 1.0
Speaker to WER: {'spk_0': 0.8698, 'spk_1': 0.8679, 'spk_2': 0.9183, 'spk_3': 0.8807, 'spk_4': 0.8512, 'spk_5': 0.8877}
Speaker clus

/home/josch080/Projektgruppe/mcorec_baseline/script/pg_utils_experiments.py:381: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.
  results_df = pd.concat([results_df, new_df], ignore_index=True)


model  avg_conv_f1  avg_speaker_wer  \
0  DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20     0.855282         0.494316   

   avg_joint_error  
0         0.331767

Saved: results_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20_by_session.csv results_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20_summary.csv


## 10 – Ergebnisanalyse: Dev-Final vs. BL4-Baseline

Robuste CSV-Suche mit Fallback (falls `exp_name` nicht gesetzt).
BL4-Identifikation unterstützt beide möglichen Spaltennamen (`model` und `exp`).

In [16]:
import glob
import pandas as pd

# Run-CSV laden
# Wenn exp_name im Notebook noch definiert ist:
run_csv = f"results_{exp_name}_by_session.csv"

# Fallback: falls exp_name nicht existiert, nimm die neueste results_*_by_session.csv
if "exp_name" not in globals():
    cands = sorted(glob.glob("results_*_by_session.csv"))
    if not cands:
        raise FileNotFoundError("Keine results_*_by_session.csv gefunden.")
    run_csv = cands[-1]
    print("exp_name nicht gesetzt -> verwende:", run_csv)
else:
    print("Verwende:", run_csv)

df_by_session = pd.read_csv(run_csv)

# Summary: Mittelwert über alle Dev-Sessions
metrics = ["avg_conv_f1","avg_speaker_wer","avg_joint_error"]

summary = (df_by_session
           .groupby("model", as_index=False)[metrics]
           .mean()
           .sort_values("avg_speaker_wer"))

summary_csv = run_csv.replace("_by_session.csv", "_summary.csv")
summary.to_csv(summary_csv, index=False)

display(summary)
print("Saved:", run_csv, summary_csv)

# BL4 Baseline laden
baseline_cands = []
baseline_cands += glob.glob("results_baseline_dev_without_central_videos*.csv")
baseline_cands += glob.glob("/mnt/data/results_baseline_dev_without_central_videos*.csv")

if not baseline_cands:
    raise FileNotFoundError("Baseline CSV nicht gefunden: results_baseline_dev_without_central_videos*.csv")

baseline_csv = sorted(baseline_cands)[0]
print("Baseline CSV:", baseline_csv)

df_base = pd.read_csv(baseline_csv)

# BL4 in beiden möglichen Spaltenformaten suchen
mask = pd.Series([False] * len(df_base))
if "model" in df_base.columns:
    mask |= df_base["model"].astype(str).eq("avsr_cocktail_finetuned")
if "exp" in df_base.columns:
    mask |= df_base["exp"].astype(str).eq("output_avsr_cocktail_finetuned")

bl4 = df_base[mask]
if bl4.empty:
    raise ValueError("BL4 nicht gefunden. Erwartet model='avsr_cocktail_finetuned' oder exp='output_avsr_cocktail_finetuned'.")
bl4 = bl4.iloc[0]

# Vergleichstabelle: Dev-Final vs. BL4
# delta_wer: negativ = Verbesserung (Run-WER < BL4-WER)
run_row = summary.iloc[0]

compare = pd.DataFrame([{
    "run_model": run_row["model"],
    "run_avg_conv_f1": float(run_row["avg_conv_f1"]),
    "run_avg_speaker_wer": float(run_row["avg_speaker_wer"]),
    "run_avg_joint_error": float(run_row["avg_joint_error"]),

    "bl4_avg_conv_f1": float(bl4["avg_conv_f1"]),
    "bl4_avg_speaker_wer": float(bl4["avg_speaker_wer"]),
    "bl4_avg_joint_error": float(bl4["avg_joint_error"]),

    "delta_wer_abs": float(run_row["avg_speaker_wer"] - bl4["avg_speaker_wer"]),
    "delta_wer_rel_%": float((run_row["avg_speaker_wer"] - bl4["avg_speaker_wer"]) / bl4["avg_speaker_wer"] * 100.0),

    "delta_joint_abs": float(run_row["avg_joint_error"] - bl4["avg_joint_error"]),
    "delta_joint_rel_%": float((run_row["avg_joint_error"] - bl4["avg_joint_error"]) / bl4["avg_joint_error"] * 100.0),
}])

compare_csv = run_csv.replace("_by_session.csv", "_vs_BL4.csv")
compare.to_csv(compare_csv, index=False)

display(compare)
print("Saved:", compare_csv)

print("\nVergleich Run vs. BL4:")
print(f"  Run: {compare.loc[0,'run_model']}")
print(f"    WER={compare.loc[0,'run_avg_speaker_wer']:.6f} | Joint={compare.loc[0,'run_avg_joint_error']:.6f}")
print(f"  BL4: avsr_cocktail_finetuned")
print(f"    WER={compare.loc[0,'bl4_avg_speaker_wer']:.6f} | Joint={compare.loc[0,'bl4_avg_joint_error']:.6f}")
print(f"  ΔWER   {compare.loc[0,'delta_wer_abs']:+.6f}  ({compare.loc[0,'delta_wer_rel_%']:+.3f}%)")
print(f"  ΔJoint {compare.loc[0,'delta_joint_abs']:+.6f}  ({compare.loc[0,'delta_joint_rel_%']:+.3f}%)")


Verwende: results_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20_by_session.csv


model  avg_conv_f1  avg_speaker_wer  \
0  DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20     0.855282         0.494316   

   avg_joint_error  
0         0.331767

Saved: results_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20_by_session.csv results_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20_summary.csv
Baseline CSV: results_baseline_dev_without_central_videos.csv


run_model  run_avg_conv_f1  \
0  DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20         0.855282   

   run_avg_speaker_wer  run_avg_joint_error  bl4_avg_conv_f1  \
0             0.494316             0.331767         0.815282   

   bl4_avg_speaker_wer  bl4_avg_joint_error  delta_wer_abs  delta_wer_rel_%  \
0             0.498727             0.354674      -0.004412        -0.884586   

   delta_joint_abs  delta_joint_rel_%  
0        -0.022906          -6.458398

Saved: results_DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20_vs_BL4.csv

Vergleich Run vs. BL4:
  Run: DEV_final_bugfix_mdOn1p0_mdOff1p2_bs12_len20
    WER=0.494316 | Joint=0.331767
  BL4: avsr_cocktail_finetuned
    WER=0.498727 | Joint=0.354674
  ΔWER   -0.004412  (-0.885%)
  ΔJoint -0.022906  (-6.458%)


## 11 – Interpretation

Der Dev-Abschlusslauf mit der aus dem Grid-Search ermittelten Konfiguration
(E72: min_on=1.0 s, min_off=1.2 s, beam=12, len=20) erzielt das
**beste Ergebnis der gesamten Experimentreihe** auf dem Dev-Set.

Die gleichen Parameter werden in `04_eval_final_results` auf das
**Eval-Set** (unbekannte Daten) angewendet, das für die Einreichung genutzt wird.

## Fazit: Finale Ergebnisse auf dem Dev-Set (25 Sessions)

| Metrik | BL4-Baseline | Dev-Final | Δ absolut | Δ relativ |
|--------|-------------|-----------|-----------|----------|
| Speaker WER (↓) | 0.4987 | 0.4943 | -0.0044 | -0.88% |
| Conv F1 (↑) | 0.8153 | 0.8553 | +0.0400 | +4.91% |
| Joint Error (↓) | 0.3547 | 0.3318 | -0.0229 | -6.46% |